services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetCredentialEmail.java

```diff
@@ -237,4 +242,16 @@ public void postInit(KeycloakSessionFactory factory) {
     public String getId() {
         return PROVIDER_ID;
     }
+
+    private boolean forceLogin(AuthenticatorConfigModel config, UserModel user) {
+        final String forceLogin = config != null? config.getConfig().get(FORCE_LOGIN) : null;
+        if (Boolean.parseBoolean(forceLogin)) {
+            return Boolean.TRUE;
+        } else if (Boolean.FALSE.toString().equalsIgnoreCase(forceLogin)) {
+            return Boolean.FALSE;
+        } else {
+            // default is only-federated
+            return !StorageId.isLocalStorage(user.getId()) || user.getFederationLink() != null;
+        }
```

Comment:
Could we be strict here to only allow the three values "true", "false", "only-federated", and throw an error if something else appears? It would make the logic simpler to grasp IMHO.

Reply:
Yes, I know this is a bit convoluted. I have changed this a bit but using `equalsIgnoreCase`, I think that admins are using any case for `true` in the configurations. Let's see if it's clearer now.